src/utils/format_open_telemetry/format_open_telemetry.cc

```diff
@@ -66,19 +66,22 @@ static void metric_to_number_data_point(NumberDataPoint *dp,
   // when we've seen a metric for the first time.
 
   // A valid metric type is guaranteed by add_metric().
-  assert(m->family->type == METRIC_TYPE_COUNTER ||
-         m->family->type == METRIC_TYPE_GAUGE);
   switch (m->family->type) {
   case METRIC_TYPE_COUNTER:
     dp->set_as_int(m->value.derive);
-    break;
+    return;
   case METRIC_TYPE_GAUGE:
+  case METRIC_TYPE_FPCOUNTER:
     dp->set_as_double(m->value.gauge);
-    break;
+    return;
   case METRIC_TYPE_UNTYPED:
-    // never reached, only here to make the compiler happy
+    // Fall through. This case signals the compiler that we're checking all
+    // values of the enum. We report an error outside of the switch to also
+    // cover other values.
     break;
   }
+  // never reached, only here to make the compiler happy
```

Comment:
IMHO this would be more appropriate in `add_metric()`, maybe just drop it from here?
```suggestion
```

Reply:
Done.